pkg/handlers/webhandler.go

```diff
@@ -76,6 +77,55 @@ func (h *AppHandler) Handle(ctx context.Context, event sdk.Event) error {
 	return nil
 }
 
+func (h *AppHandler) reconcile(cr *v1alpha1.WebApp) error {
+	//reconcile template params into deployment config
+	dc, err := h.osClient.GetDC(cr.Namespace, "tutorial-web-app")
+	if err != nil {
+		return err
+	}
+	dcUpdated := false
+	for _, param := range []string{"OPENSHIFT_OAUTHCLIENT_ID", "OPENSHIFT_HOST", "SSO_ROUTE", "WALKTHROUGH_LOCATIONS"} {
+		if val, ok := cr.Spec.Template.Parameters[param]; ok {
+			dcUpdated, dc.Spec.Template.Spec.Containers[0] = updateOrCreateEnvVar(dc.Spec.Template.Spec.Containers[0], param, val)
+		} else {
+			//key does not exist in CR, ensure it is not present in the DC
+			dcUpdated, dc.Spec.Template.Spec.Containers[0] = deleteEnvVar(dc.Spec.Template.Spec.Containers[0], param)
+		}
+	}
+	//update the DC
+	if dcUpdated {
+		return h.osClient.UpdateDC(cr.Namespace, &dc)
+	}
+	return nil
+}
+
+func deleteEnvVar(container corev1.Container, envName string) (bool, corev1.Container) {
+	for k, envVar := range container.Env {
```

Comment:
**nit** Not sure how much I like having a function for this when only being used in one place.

Reply:
I prefer this approach because:
- It's easier to compile the reconcile function into your head
- It allows me to avoid using break and use a return instead
- It doesn't cost extra to split it out this way